tests/blueprint_with_behavior.ts

```diff
@@ -0,0 +1,93 @@
+export const bpBuilding1 = blueprint.building2x1_2M_1({
+    name: "Bot Factory",
+    medium: [
+        component.roboticsAssembler([
```

Comment:
Is there a reason these need to be passed in an array?

Reply:
I agree about javascript overloading, but we'd be resolving at compile time so there would be no runtime cost.

Reply:
No, just compile time complexity of figuring out what the user meant which is what I was referring to. Sure, we can do it. But I'm not a fan of building an API around overloading when it makes the implementation much harder and the usage remains roughly equivalent.

I'm not against it, it just doesn't have my preference.